packages/client/hmi-client/src/services/calibrate-workflow.ts

```diff
@@ -135,31 +135,40 @@ export const autoCalibrationMapping = async (
 	});
 	// Get all data column groundings
 	datasetOptions.forEach((col) => {
-		const dataGroundingList = Object.keys(col.metadata?.groundings?.identifiers);
-		dataGroundingList.forEach((ele) => allDataGroundings.push(ele));
+		if (col.metadata?.groundings?.identifiers) {
+			Object.entries(col.metadata?.groundings?.identifiers)
+				.map((ele) => ele.join(':'))
+				.forEach((ele) => allDataGroundings.push(ele));
+		}
 	});
 
 	// take out duplicates:
 	const distinctModelGroundings = [...new Set(allModelGroundings)];
 	const distinctDataGroundings = [...new Set(allDataGroundings)];
 	const allSimilarity = await getEntitySimilarity(distinctModelGroundings, distinctDataGroundings);
 	if (!allSimilarity) return result;
-
-	const filteredSim = allSimilarity.filter((ele) => ele.distance < acceptableDistance);
+	// normalized cosine similarities
```

Comment:
I bet you never thought you would write such a comment in your life!?

Reply:
LOOL very true